Add hardware profile table

DaoDaoNoCode opened the pull request:
<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
<!--- All code change PRs should relate to an issue, reference it here; see example below -->
<!--- https://issues.redhat.com/browse/RHOAIENG-123456 -->
JIRA: https://issues.redhat.com/browse/RHOAIENG-16254

## Description
<!--- Describe your changes in detail; the what, the why, any findings, etc -->
<!--- Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->
Added the table for the hardware profiles.

Main table:
<img width="1512" alt="Screenshot 2024-12-12 at 3 59 24 PM" src="https://github.com/user-attachments/assets/611932c8-fe74-481e-86fd-0bb681a1fb24" />

Expandable sections:
There are 3 tables inside, if there is no data in the table, the corresponding nested table will be hidden.
<img width="1512" alt="Screenshot 2024-12-12 at 4 00 13 PM" src="https://github.com/user-attachments/assets/37a9acaf-817a-4c98-9b47-0231800311e8" />

Filters:
You can filter by Name and Enable columns.
<img width="1512" alt="Screenshot 2024-12-12 at 4 01 27 PM" src="https://github.com/user-attachments/assets/82d9ac8e-a2e1-4673-aee6-4d14e4a96349" />
<img width="1512" alt="Screenshot 2024-12-12 at 4 02 05 PM" src="https://github.com/user-attachments/assets/16f2647b-3f36-4a2b-afc5-c3da3d7ed12e" />

Disable hardware profile modal:
<img width="1512" alt="Screenshot 2024-12-12 at 4 03 00 PM" src="https://github.com/user-attachments/assets/30f311e3-25c2-4c18-8ab3-710f1d1424fe" />

Delete modal:
<img width="1512" alt="Screenshot 2024-12-12 at 4 03 20 PM" src="https://github.com/user-attachments/assets/0c300c41-339b-4233-9dd0-f3bff9883914" />

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
1. Create some hardware profiles CRs on the clusters in the `opendatahub` namespace
2. Try to set the empty array to `nodeSelectors`, `identifiers` and `tolerations` fields in `spec` to see how the expandable section shows the tables
3. Try to toggle the enable switch to see the modal and how it performs
4. Try the delete function
5. Try the filters, make sure it can filter by the name and enable status

## Test Impact
<!--- What tests have you done to covert implemented functionality -->
<!--- If tests are not applicable, explain why here -->
Added cypress tests for this page.

## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

DaoDaoNoCode commented:
@vconzola Hi, can you check the screenshots?

DaoDaoNoCode commented on frontend/src/pages/hardwareProfiles/const.ts:
@dpanshug I prefer to keep everything in the root `const.ts` file because we use different names for the column variables. Just let you know in advance so it's easier for you to rebase if needed.

vconzola commented:
@DaoDaoNoCode A couple comments questions...

1. What font size are you using for the section headers (Node resources, Node selectors, Tolerations)? They look kinda big.
2. We should not require confirmation when disabling a profile. We don't confirm when disabling other resources, like notebook images and serving runtimes.
3. Let's change the column header to "Enabled" instead of "Enable".
4. All of the content will need a review by @kaedward. Do you want to do that now or after the code is merged?

DaoDaoNoCode commented:
@vconzola Hi, thanks for the feedback.
1. I am using `h6` variant for `Title` component, which I believe size is `md` here. This is the smallest size of the `Title` component, I can change it to normal Text and add bold style manually if you think it's too big now.
2. I can remove this disabled modal. I saw it when disabling the accelerator profiles so I just copied what they do
<img width="1512" alt="Screenshot 2024-12-16 at 9 20 15 AM" src="https://github.com/user-attachments/assets/a3d9c7fe-ca04-4524-8969-1c6ff4d33115" />
3. Will do
4. I prefer to get the content reviewed after all the codes are merged

DaoDaoNoCode commented:
@vconzola I addressed the feedback as suggested. Can you check it again?

<img width="1512" alt="Screenshot 2024-12-16 at 9 49 59 AM" src="https://github.com/user-attachments/assets/eff1678c-114f-47df-a8f1-b0386ab03577" />

vconzola commented:
Thanks @DaoDaoNoCode This LGTM. I think it looks better with the smaller font for the section headers. Regarding the confirmation dialog, leave it out for now. I've started a thread within the UXD channel to talk about this, because we're inconsistent in different parts of the UI currently. I'll let you know what we decide, but I don't think that should block you from getting your PR merged, right?

Gkrumbach07 reviewed (approved):
/lgtm

looks great
